### PR TITLE
343836 application submitted page

### PR DESCRIPTION
**Pull request by AtosSK**

### CODE REVIEW

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 1]
```diff
@@ -210,5 +210,40 @@ public async Task<IActionResult> GetLegalDocumentsAddress(Guid id)
...
+
+            return Ok(accreditation.ReferenceNumber);
+        }
+
+
+        [HttpGet("Accreditation")]
+        public async Task<IActionResult> GetAccredition(
+            Guid id)
+        {
+            var accreditation = await _accreditationService.GetAccrediation(id);
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:251]: Do you need to pass an entire accreditation class back to Portal? This is the same DTO as was passed from API to Facade. We should really be passing only the properties we need in Portal. Therefore there should be a seperate DTO class under the Portal folder in DTOs, to help create that seperation

Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:251]: Done.

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 2]
```diff
@@ -210,5 +210,39 @@ public async Task<IActionResult> GetLegalDocumentsAddress(Guid id)
 
             return Ok();
         }
+
+        /// <summary>
+        /// Updates the the reference number for the accrediation.
+        /// </summary>
+        /// <param name="id"></param>
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs]: No description for the variables

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 3]
```diff
@@ -210,5 +210,39 @@ public async Task<IActionResult> GetLegalDocumentsAddress(Guid id)
...
+            Guid id,
+            [FromBody] string referenceNumber)
+        {
+            await _accreditationService.UpdateReferenceNumber(id, referenceNumber);
+
+            return Ok();
+        }
+
+        [HttpGet("ReferenceNumber")]
+        public async Task<IActionResult> GetReferenceNumber(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:234]: No method description

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 4]
```diff
@@ -210,5 +210,39 @@ public async Task<IActionResult> GetLegalDocumentsAddress(Guid id)
...
+            Guid id)
+        {
+            var accreditation = await _accreditationService.GetAccrediation(id);
+
+            return Ok(accreditation.ReferenceNumber);
+        }
+
+        [HttpGet("AccreditationFee")]
+        public async Task<IActionResult> GetAccreditationFee(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:248]: Method description

File: src/EPR.Accreditation.Facade/Services/AccreditationService.cs [hunk 5]
```diff
@@ -339,5 +340,22 @@ public async Task<AddressDto> GetLegalDocumentsAddress(Guid id)
                 id,
                 accreditation);
         }
+
+        public async Task UpdateReferenceNumber(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/AccreditationService.cs:356]: No method descriptions

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs (hunk 2 above)
Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs]: Done.

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs (hunk 4 above)
Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:248]: Done.

File: src/EPR.Accreditation.Facade/Services/AccreditationService.cs (hunk 5 above)
Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Services/AccreditationService.cs:356]: Done.

File: src/EPR.Accreditation.Facade/Services/AccreditationService.cs [hunk 6]
```diff
@@ -339,5 +342,45 @@ public async Task<AddressDto> GetLegalDocumentsAddress(Guid id)
...
+            return await _httpAccreditationService.GetAccreditation(id);
+        }
+
+
+        /// <summary>
+        /// Builds a random string.
+        /// </summary>
+        /// <param name="length">Lenght of desired string.</param>
+        /// <returns>The random string.</returns>
+        private string RandomString(int length)
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/AccreditationService.cs]: I don't think this should be here any longer should it?

Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Services/AccreditationService.cs]: Done.

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**